conan/tools/google/bazeldeps.py

```diff
@@ -1,278 +1,583 @@
 import os
-import platform
+import re
 import textwrap
+from collections import namedtuple
 
-from jinja2 import Template
+from jinja2 import Template, StrictUndefined
 
+from conan.errors import ConanException
 from conan.tools._check_build_profile import check_using_build_profile
-from conans.errors import ConanException
 from conans.util.files import save
 
+_DepInfo = namedtuple("DepInfo", ['name', 'requires', 'cpp_info'])
+_LibInfo = namedtuple("LibInfo", ['name', 'is_shared', 'lib_path', 'interface_lib_path'])
+
+
+def _get_name_with_namespace(namespace, name):
+    """
+    Build a name with a namespace, e.g., openssl-crypto
+    """
+    return f"{namespace}-{name}"
+
+
+def _get_package_reference_name(dep):
+    """
+    Get the reference name for the given package
+    """
+    return dep.ref.name
+
+
+def _get_package_name(dep):
+    pkg_name = dep.cpp_info.get_property("bazel_module_name") or _get_package_reference_name(dep)
```

Comment:
I think we should probably have a property to define each part of the label that's used for the dependencies. For example, if we have this label: `@name1//:name2`, have:
- `bazel_repository_name` to define the `name1` part
- `bazel_target_name` to define the `name2` part 
Instead of just 1 property that would define both `name1` and `name2` for the root component and just name2 for other components
Also, we may want to check that the `bazel_repository_name` is not set from the components, just from the root cpp_info

Reply:
Agree 😄